# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../nginx/legacy_sus.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

usleep() was deprecated by SUSv3 and newer LIBCs are starting to enforce that
deprecation.

--- ./src/os/unix/ngx_time.h.orig	2010-09-24 16:16:23.000000000 -0400
+++ ./src/os/unix/ngx_time.h	2010-09-24 16:22:40.000000000 -0400
@@ -58,7 +58,7 @@
 void ngx_libc_gmtime(time_t s, struct tm *tm);
 
 #define ngx_gettimeofday(tp)  (void) gettimeofday(tp, NULL);
-#define ngx_msleep(ms)        (void) usleep(ms * 1000)
+#define ngx_msleep(ms)        (void) nanosleep(&((struct timespec){0,(ms) * 1000000ul}), NULL)
 #define ngx_sleep(s)          (void) sleep(s)
 
 
